sync/src/chain.rs

```diff
@@ -1274,7 +1273,7 @@ impl ChainSync {
 		let pv64 = io.eth_protocol_version(peer) >= 64;
 		let mut packet = RlpStream::new_list(if pv64 { 7 } else { 5 });
 		let chain = io.chain().chain_info();
-		packet.append(&(PROTOCOL_VERSION as u32));
+		packet.append(&(io.eth_protocol_version(peer) as u32));
```

Comment:
digging into the code, seems that `eth_protocol_version` is a fairly expensive call. maybe keep the result of the first call?